### PR TITLE
[4.8.x] Add libxml2

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Review by ocefpaf — APPROVED** [review]
Thanks for backporting this!

**Comment by xylar** [comment]
Thanks again to @jschueller!

**Comment by xylar** [comment]
Hmm, I need to work on this some more tomorrow...

File: recipe/patches/fix-tests.patch [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
++++ netcdf-c-4.8.1-patch/nc_test/CMakeLists.txt	2022-10-30 12:00:09.828975244 +0100
+@@ -94,11 +94,6 @@
+       add_sh_test(nc_test run_diskless2)
+     ENDIF()
+ 
+-    IF(ENABLE_BYTERANGE)
+-        build_bin_test_no_prefix(tst_byterange)
+-        add_sh_test(nc_test test_byterange)
+-    ENDIF()
+-
```
Comment (xylar) [inline recipe/patches/fix-tests.patch:44]: Disabling test `nc_test_test_byterange` that is failing because of an inaccessible server.

**Pull request merged**